zkvm/src/constraints.rs

```diff
@@ -159,6 +208,14 @@ impl Add for Expression {
     }
 }
 
+impl Sub for Expression {
```

Comment:
I don't think we need impl of `Sub`. For `eq` instruction we need a different type `Constraint::Eq(Expr, Expr)`.